app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -11,27 +10,49 @@ def self.call(*args)
       end
 
       def call
-        return if commented_on_welcome_thread? || received_notification?
+        return unless user
+
+        send_welcome_notification
+        send_authentication_notification
+      end
+
+      def send_welcome_notification
+        return if received_notification?(welcome_broadcast) || commented_on_welcome_thread?
```

Comment:
If we're going to take the approach of adding a check for `user.created_at` on line `26`, then we need to add a check in this notification (I think we want to send the welcome notification 3 hours after `user.created_at`. I know that this code is not related to the feature but we should be sure to add that in this PR or the next one so that we don't forget about it :)